deep_quoridor/src/agents/alphazero.py

```diff
@@ -30,6 +37,119 @@ class AlphaZeroParams(SubargsBase):
     # Exploration vs exploitation.  0 is pure exploitation, infinite is random exploration.
     temperature: float = 1.0
 
+    # Number of searches
+    n: int = 1000
+    # A higher number favors exploration over exploitation
+    c: float = 1.4
+
+
+class AzNode:
+    def __init__(
+        self,
+        game: Quoridor,
+        parent: Optional["AzNode"] = None,
+        action_taken: Optional[Action] = None,
+        ucb_c: float = 1.0,
+        prior: float = 0.0,
+    ):
+        self.game = game
+        self.parent = parent
+        self.action_taken = action_taken
+
+        self.children = []
+        self.visit_count = 0
+        self.value_sum = 0.0
+
+        self.ucb_c = ucb_c
+        self.prior = prior
+
+    def should_expand(self):
+        return len(self.children) == 0
+
+    def expand(self, policy_probs: np.ndarray):
+        """
+        Create a child of the current node.
```

Comment:
Here you create all the childs, right? Maybe update the comment to match?

Also consider if you wish, just to keep both evolving in parallel to make them easier to compare, to also update mcts.py to do it in this same way to match. If I understand things correctly, the default behavior would be to expand all with equal probability, maybe? I am probably missing something. Just trying to use the opportunity to understand a bit more, that's all.

Reply:
We could change the other MCTS the same way to expand all the nodes (although besides code re-using, I don't think it provides any advantage).   In that one,  it doesn't use priors, so it's an uniform probability.  Basically the advantage here is that it uses the "intuition" of the NN to start exploring the nodes that it estimates are more likely to get to a win.
I'll add a TODO for myself to do the update in the other one